lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -101,79 +90,49 @@ UsdSceneItem::Ptr UsdUndoRenameCommand::renamedItem() const
 
 bool UsdUndoRenameCommand::renameRedo()
 {
-    // Copy the source path using CopySpec, and remove the source.
-
-    // We use the source layer as the destination.  An alternate workflow
-    // would be the edit target layer be the destination:
-    // _layer = _stage->GetEditTarget().GetLayer()
-    bool status = SdfCopySpec(_layer, _usdSrcPath, _layer, _usdDstPath);
-    if (status) {
-        // remove all scene description for the given path and 
-        // its subtree in the current UsdEditTarget 
-        {
-            UsdEditContext ctx(_stage, _layer);
-            status = _stage->RemovePrim(_ufeSrcItem->prim().GetPath());
-        }
+    auto primSpec = MayaUsdUtils::getPrimSpecAtEditTarget(_stage, _prim);
 
-        if (status) {
-            // The renamed scene item is a "sibling" of its original name.
-            _ufeDstItem = createSiblingSceneItem(_ufeSrcItem->path(), _usdDstPath.GetElementString());
+    if(primSpec) {
 
-            sendRenameNotification(_ufeDstItem, _ufeSrcItem->path());
-        }
+        // set prim's name
+        primSpec->SetName(_newName);
+
+         // the renamed scene item is a "sibling" of its original name.
+        _ufeDstItem = createSiblingSceneItem(_ufeSrcItem->path(), _newName);
+
+        sendRenameNotification(_ufeDstItem, _ufeSrcItem->path());
     }
     else {
-        UFE_LOG(std::string("Warning: SdfCopySpec(") +
-                _usdSrcPath.GetString() + std::string(") failed."));
+        return false;
     }
 
-    return status;
+    return true;
```

Comment:
On Redo:
1- Get the SdfPrimSpec to the _prim and check it's validity.
2- Set it's name to the new name: primSpec->SetName(_newName);
3- Set _ufeDstItem and send UFE scene notification